From: Jiri Pirko <jpirko@redhat.com>
Date: Tue, 25 Jan 2011 12:53:50 -0500
Subject: [net] core: clear allocs for privileged ethtool actions
Message-id: <20110125125349.GA2681@psychotron.brq.redhat.com>
Patchwork-id: 32859
O-Subject: [RHEL5.7 patch] BZ672433 CVE-2010-4655 net: clear heap allocations
	for privileged ethtool actions
Bugzilla: 672433
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Jiri Olsa <jolsa@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Thomas Graf <tgraf@redhat.com>

BZ672433
https://bugzilla.redhat.com/show_bug.cgi?id=672433

Upstream:
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=b00916b189d13a615ff05c9242201135992fcda3

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=3061125

Jirka

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/core/ethtool.c b/net/core/ethtool.c
index 7824d0d..e4f55db 100644
--- a/net/core/ethtool.c
+++ b/net/core/ethtool.c
@@ -183,7 +183,7 @@ static int ethtool_get_regs(struct net_device *dev, char __user *useraddr)
 	if (regs.len > reglen)
 		regs.len = reglen;
 
-	regbuf = kmalloc(reglen, GFP_USER);
+	regbuf = kzalloc(reglen, GFP_USER);
 	if (!regbuf)
 		return -ENOMEM;
 
